### PR TITLE
refactor(typescript): migrate label

### PR DESCRIPTION
**Pull request by priyajeet**

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @priyajeet has signed the CLA. Thanks for the pull request!

File: src/components/link/stories/LinkGroup.stories.tsx [hunk 1]
```diff
@@ -7,6 +7,7 @@ import notes from './LinkGroup.stories.md';
 
 export const basic = () => (
     <LinkGroup>
+        <button type="button">foo</button>
```
Comment (j3tan) [inline src/components/link/stories/LinkGroup.stories.tsx]: intentional?

Comment (priyajeet) [reply src/components/link/stories/LinkGroup.stories.tsx]: good catch, I was testing if TS complains about it. It doesn't :(

File: src/components/label/HiddenLabel.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import * as React from 'react';
+
+import LabelPrimitive from './LabelPrimitive';
+
+export interface HiddenLabelProps {
+    /** Child for the label */
+    children: React.ReactElement;
+    /** Text content of the label */
+    labelContent: React.ReactNode;
```
Comment (sahiga) [inline src/components/label/HiddenLabel.tsx]: Should this be `React.ReactText`?

File: src/components/label/Label.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    /** Optional props for the icon */
+    infoIconProps?: Record<string, any>; // eslint-disable-line @typescript-eslint/no-explicit-any
+    /** Tooltip text for the info icon */
+    infoTooltip?: React.ReactNode;
+    /** Whether to show the `(Optional)` text next to the label for an optional field */
+    showOptionalText?: boolean;
+    /** The label text */
+    text: React.ReactNode;
+    /** Optional tooltip text for the label */
+    tooltip?: React.ReactNode;
```
Comment (sahiga) [inline src/components/label/Label.tsx]: Should `text` and `tooltip` also be `React.ReactText`?

File: src/components/label/HiddenLabel.tsx (hunk 2 above)
Comment (priyajeet) [reply src/components/label/HiddenLabel.tsx]: `<FormattedMessage>` is a node

File: src/components/label/Label.tsx (hunk 3 above)
Comment (priyajeet) [reply src/components/label/Label.tsx]: `<FormattedMessage>` is a node

**Pull request merged**